### ISSUE TITLE
Using Preferred Support Class Not Working

### ISSUE THREAD

**Issue by Rajvir8** [29988122/Fate-Grand-Order_Lua#66]
It's probably because I took the picture wrong but I did a couple attempts and I got nothing, I would put it in preferred mode and tell it to look for the PNG file I named Bond.

I made sure to resize it using the tool given.

But after waiting a minute in support mode nothing happened.

I also thought it should be updating the support screen if it couldn't find it but nothing happened?

I changed it to

Support_SelectionMode = "preferred"
Support_PreferredImage = "Bond.png"
Support_SwapsPerRefresh = 10
Support_MaxRefreshes = 3
Support_FallbackTo = "first"

But nothing ever happened?

Any ideas?

For now I'll swap back to using first.

![bond](https://user-images.githubusercontent.com/17600697/45359408-73f7a280-b59a-11e8-8ff3-dc2ecdb916cd.PNG)

**Comment by potchy** [comment]
Hi, would you mind submitting a screenshot of your support screen?
Also, don't resize the image. It will make it harder for the scanner to find it.

**Comment by 29988122** [comment]
@potchy 

Currently the compare dimension is 1280*720 (720p, suggested by Ankulua Author), script dimension is 2560*1440 (2K 16:9, I developed that originally on 2K screen phone).

If the picture was directly cropped from a 2560*1440 screenshot, will it be detected correctly by the script? Just curious, because I resize it to 1/2 (1440/2 = 720) personally to fit the compare dimension every time...

If so, I might need to edit readme.md, since I got it wrong completely.

**Comment by potchy** [comment]
No, you're right, my bad.
However, if you do resize images for the script, it might be best to do so without interpolation, though.

**Comment by ryuga93** [comment]
I think the scaling should be fine as long as the ratio is 16:9

**Comment by Rajvir8** [comment]
Sorry for the delay in replying I ended up fairly busy.

It might be because I'm using Nox, what I ended up doing is using the snipping tool to snap the portion I wanted and then resizing it.

But no the script didn't run at all, maybe it's because I started it in the friend menu rather than let it open it?

**Comment by Rajvir8** [comment]
It's currently on my Phone as the Nero event is hard enough I can't rely on autoing while having 6 petals, but I'll put it back on Nox and start using the script after the event's over.

**Comment by Rajvir8** [comment]
I'll see about retrying to see if the problem is still there, if it is I'll take a screenshot and censor the names as I'm not sure how I'd do a video while censoring it.

**Comment by potchy** [comment]
You're supposed to start the script in the quest listing or during battle.

**Comment by Rajvir8** [comment]
I updated the script and Nox, I then tried it from the menu with a new Support Picture.

The new support Picture friendship works but the old Bond one still doesn't work.

It might be because I resized it and it doesn't fit my screen anymore or some other reason.

Glad that I got it working now though.

It took a couple of attempts but I also got Bond working.

I can also swap between them by creating an Additional Folder named Changing Support, and a sub folder for each one I want.

Then I can just copy paste that notepad to the Anka Lua.

I need that addtional step as it can't be edited and saved without Root, and Fate Grand Order can't run without root so this way I don't have to do turn Nox on and off twice just to change it.
